lib/SimpleSAML/SessionHandlerPHP.php

```diff
@@ -77,13 +77,28 @@ protected function __construct()
         $params = $this->getCookieParams();
 
         if (!headers_sent()) {
-            session_set_cookie_params(
-                $params['lifetime'],
-                $params['path'],
-                $params['domain'],
-                $params['secure'],
-                $params['httponly']
-            );
+            if (\PHP_VERSION_ID >= 70300) {
+                session_set_cookie_params([
+                    'lifetime' => $params['lifetime'],
+                    'path' => $params['path'],
+                    'domain' => $params['domain'],
+                    'secure' => $params['secure'],
+                    'httponly' => $params['httponly'],
+                    'samesite' => $params['samesite'],
```

Comment:
This does not seem to work on PHP 7.3. According to the docs, "The value of the samesite element should be either Lax or Strict." and maybe somewhat surprisingly the docs seem to match reality: None is not accepted. So maybe also for 7.3 we should use the below strategy (also in the other invocations of this and of setcookie).

Reply:
We tested this against 7.3.6 and found it did work. I also looked at the PHP commit that introduced this, and there's no validation of the string in it. (On the other hand, the hacky version definitely does *not* work for PHP >= 7.3)

Reply:
I tested this also with 7.3.6 and reached my conclusion (changing it to `Lax` did cause the attribute to appear). Will investigate further then.

Reply:
My unit test still isn't quite right (it's checking an absolute string value whereas the formatting varies per version of PHP). Nevertheless, even the failing unit test shows the correct values for 7.3: https://travis-ci.org/simplesamlphp/simplesamlphp/jobs/554825414#L647

Reply:
This is now explicitly unit tested. The `session_set_cookie_params` from this comment is tested at https://github.com/simplesamlphp/simplesamlphp/pull/1153/commits/4d341015e4a2f736652a08dd0685438ac7071279#diff-beeb6dbefe9cf458457456e4cb21a72dR80 and `setcookie` is tested at https://github.com/simplesamlphp/simplesamlphp/pull/1153/commits/5bab223804d7a518ae7ec95077899964d617ece9#diff-8af11dde64458ba57910e950558388ceR498

Reply:
Thanks. It does indeed work for me. It's Firefox' inspector that doesn't show the unknown value.